lib/proxy/router.go

```diff
@@ -114,6 +120,8 @@ type RouterConfig struct {
 	ClusterName string
 	// Log is the logger to use
 	Log *logrus.Entry
+	// Emitter is an event emitter
```

Comment:
Can you mention why `Emitter` is needed and when/how it is being used? Also, why is it needed right now?